cache/rediscache/redicache_test.go

```diff
@@ -0,0 +1,204 @@
+package rediscache_test
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/terranodo/tegola/cache"
+	"github.com/terranodo/tegola/cache/rediscache"
+)
+
+// TestNew will run tests against a local redis instance
+// on 127.0.0.1:6379
+func TestNew(t *testing.T) {
+	if os.Getenv("RUN_REDIS_TESTS") != "yes" {
+		fmt.Println("RUN_REDIS_TESTS not set to 'yes' skipping tests")
+		return
+	}
+
+	type tc struct {
+		config map[string]interface{}
+		err    error
+	}
+
+	testcases := map[string]tc{
+		"redis explicit config": {
+			config: map[string]interface{}{
+				"network":  "tcp",
+				"address":  "127.0.0.1:6379",
+				"password": "",
+				"db":       0,
+				"max_zoom": 0,
+			},
+			err: nil,
+		},
+		"redis implicit config": {
+			config: map[string]interface{}{},
+			err:    nil,
+		},
+		"redis bad config":{
+			config: map[string]interface{}{
+				"address": "127.0.0.1:6000",
+			},
+			err: fmt.Errorf("dial tcp 127.0.0.1:6000: connect: connection refused"),
+		},
+	}
+
+	for i, tc := range testcases {
+		_, err := rediscache.New(tc.config)
+		if err != nil {
+			if tc.err != nil && err.Error() == tc.err.Error() {
+				//	correct error returned
+				continue
+			}
+			t.Errorf("[%v] unexpected err, expected %v got %v", i, tc.err, err)
+			continue
+		}
+	}
+}
+
+func TestSetGetPurge(t *testing.T) {
+	if os.Getenv("RUN_REDIS_TESTS") != "yes" {
+		return
+	}
+
+	type tc struct {
+		config       map[string]interface{}
+		key          cache.Key
+		expectedData []byte
+		expectedHit  bool
+	}
+
+	testcases := map[string]tc {
+		"redis cache hit": {
+			config: map[string]interface{}{},
+			key: cache.Key{
+				Z: 0,
+				X: 1,
+				Y: 2,
+			},
+			expectedData: []byte("\x53\x69\x6c\x61\x73"),
+			expectedHit:  true,
+		},
+		"redis cache miss": {
+			config: map[string]interface{}{},
+			key: cache.Key{
+				Z: 0,
+				X: 0,
+				Y: 0,
+			},
+			expectedData: []byte{},
+			expectedHit:  false,
+		},
+	}
+
+	for k, tc := range testcases {
+		rc, err := rediscache.New(tc.config)
+		if err != nil {
+			t.Errorf("[%v] unexpected err, expected %v got %v", k, nil, err)
+			continue
+		}
+
+		//	test write
+		if tc.expectedHit {
+			err = rc.Set(&tc.key, tc.expectedData)
+			if err != nil {
+				t.Errorf("[%v] unexpected err, expected %v got %v", k, nil, err)
+			}
+			continue
+		}
+
+		// test read
+		output, hit, err := rc.Get(&tc.key)
+		if err != nil {
+			t.Errorf("[%v] read failed with error, expected %v got %v", k, nil, err)
+			continue
+		}
+		if tc.expectedHit == hit {
```

Comment:
Should this be `tc.expectedHit != hit` ?

Reply:
Yes it should